pkg/tsdb/azuremonitor/httpclient.go

```diff
@@ -32,6 +32,11 @@ func httpClientProvider(route azRoute, model datasourceInfo, cfg *setting.Cfg) (
 func newHTTPClient(route azRoute, model datasourceInfo, cfg *setting.Cfg) (*http.Client, error) {
 	model.HTTPCliOpts.Headers = route.Headers
 
+	// Inject API-Key for AppInsights
+	if _, ok := model.DecryptedSecureJSONData["appInsightsApiKey"]; ok {
+		model.HTTPCliOpts.Headers["X-API-Key"] = model.DecryptedSecureJSONData["appInsightsApiKey"]
+	}
+
```

Comment:
You create a separate instance of http client for each DatasourceService, can you create http client with either API key or Azure Token provider depending on the service?

Reply:
good point. I still have to apply Markus feedback and reuse the http client provider that already exists but I will move this to a middleware.